packages/nimbus/src/components/primary-action-dropdown/primary-action-dropdown.tsx

```diff
@@ -0,0 +1,347 @@
+      onAction(buttonContent.actionId);
+    }
+  };
+
+  const isPrimaryDisabled = isDisabled || buttonContent.isDisabled;
+
+  // In regular mode, if there are no actionable items, the entire button should be disabled
+  // In split mode, only disable the dropdown trigger if there are no actionable items
+  const isDropdownTriggerDisabled = isDisabled || !hasActionableItems;
+  const isRegularButtonDisabled = isSplitButtonMode
+    ? isPrimaryDisabled
+    : isDropdownTriggerDisabled;
+
+  return (
+    <PrimaryActionDropDownRootSlot
+      variant={variant}
+      data-mode={isSplitButtonMode ? "split" : "regular"}
+    >
+      {isSplitButtonMode ? (
+        // Split Button Mode: Separate primary button + dropdown trigger
+        <PrimaryActionDropDownButtonGroupSlot>
+          {/* Primary Action Button - Internal to component */}
+          <PrimaryActionDropDownPrimaryButtonSlot asChild>
+            <Button
+              {...buttonProps}
+              isDisabled={isPrimaryDisabled}
+              onPress={executePrimaryAction}
+            >
+              {iconElement}
+              {buttonContent.content}
+            </Button>
+          </PrimaryActionDropDownPrimaryButtonSlot>
+
+          {/* Menu Trigger and Dropdown */}
+          <Menu.Root
+            trigger="press"
+            isOpen={isOpen}
+            defaultOpen={defaultOpen}
+            onOpenChange={onOpenChange}
+            placement="bottom start"
+            selectionMode="none"
+            onAction={onAction ? (key) => onAction(String(key)) : undefined}
+          >
+            <Menu.Trigger asChild>
+              <PrimaryActionDropDownTriggerSlot asChild>
+                <IconButton
+                  {...buttonProps}
+                  aria-label={ariaLabel}
+                  isDisabled={isDropdownTriggerDisabled}
+                >
+                  <KeyboardArrowDown />
+                </IconButton>
+              </PrimaryActionDropDownTriggerSlot>
+            </Menu.Trigger>
+
+            <Menu.Content>{menuItems}</Menu.Content>
+          </Menu.Root>
+        </PrimaryActionDropDownButtonGroupSlot>
+      ) : (
+        // Regular Button Mode: Single button that opens dropdown on click
+        <Menu.Root
+          trigger="press"
+          isOpen={isOpen}
+          defaultOpen={defaultOpen}
+          onOpenChange={onOpenChange}
+          placement="bottom start"
+          selectionMode="none"
+          onAction={onAction ? (key) => onAction(String(key)) : undefined}
+        >
+          <Menu.Trigger asChild>
+            <PrimaryActionDropDownPrimaryButtonSlot asChild>
+              <Button
+                {...buttonProps}
+                isDisabled={isRegularButtonDisabled}
+                aria-label={ariaLabel}
+              >
+                {iconElement}
+                {buttonContent.content}
+                <KeyboardArrowDown />
+              </Button>
```

Comment:
It was _so nice_ to leverage our own components internally and not have to reinvent the wheel. Unfortunately for me, I discovered this only after spending time reinventing all the wheels